lib/src/named_sections/named_section.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:meta/meta.dart';
+
+@immutable
+class NamedSection {
+  /// Zero-based index of the line with the starting tag.
+  final int startLine;
+
+  /// Zero-based index of the line with the ending tag.
+  final int endLine;
+
```

Comment:
is the space between declarations needed?

Reply:
Blank lines separate doc comment + declaration pairs and improve doc comment clarity.
Declarations without doc comments are OK without blank lines between them.